src/flint/fmpq.jl

```diff
@@ -1205,14 +1205,20 @@ Rational(z::ZZRingElem) = Rational{BigInt}(z)
 
 ###############################################################################
 #
-#   Convenience methods for arithmetics (since `QQFieldElem` is not a `Number` type)
+#   Convenience methods for arithmetics (since `QQFieldElem` and `ZZRingElem` are not `Number` types)
 #
 ###############################################################################
 
 //(v::Vector{QQFieldElem}, x::QQFieldElem) = v .// x
+/(v::Vector{QQFieldElem}, x::QQFieldElem) = v ./ x
 *(x::QQFieldElem, v::Vector{QQFieldElem}) = x .* v
 *(v::Vector{QQFieldElem}, x::QQFieldElem) = v .* x
 
+//(v::Vector{QQFieldElem}, x::ZZRingElem) = v .// x
+/(v::Vector{QQFieldElem}, x::ZZRingElem) = v ./ x
+*(x::ZZRingElem, v::Vector{QQFieldElem}) = x .* v
+*(v::Vector{QQFieldElem}, x::ZZRingElem) = v .* x
```

Comment:
I think next people will ask to do the same for a vector polynomials etc. So perhaps we should right away do this:
```suggestion
//(v::Vector{<:NCRingElem}, x::ZZRingElem) = v .// x
/(v::Vector{<:NCRingElem}, x::ZZRingElem) = v ./ x
*(x::ZZRingElem, v::Vector{<:NCRingElem}) = x .* v
*(v::Vector{<:NCRingElem}, x::ZZRingElem) = v .* x
```

Reply:
No, this will open up the `Union{}[]` ambiguity. There is a workaround, but it is a bit complicated. We should just merge this here as is.

Reply:
Ah, maybe not, sorry.

Reply:
So commit Max's suggestion and wait for the tests to run through?

Reply:
Yeah, but it's too late to close the door after the horse has escaped the barn.

Reply:
OK, then commit. Worst case some unforeseen problems pop up and this PR is delayed. Your call.

Reply:
That should probably be in AbstractAlgebra, where NCRingElem is defined.